src/Middleware/SortParser.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Choredo\Middleware;
+
+use Assert\Assert;
+use Choredo\Sortable;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use const Choredo\REQUEST_HANDLER_CLASS;
+use const Choredo\REQUEST_SORT;
+
+class SortParser
+{
+    public function __invoke(Request $request, Response $response, callable $next)
+    {
+        /** @var Sortable $handler */
+        $handler = $request->getAttribute(REQUEST_HANDLER_CLASS);
+        if (!in_array(Sortable::class, class_implements($handler))) {
+            return $next($request, $response);
+        };
+
+        $sort = $request->getQueryParams()[REQUEST_SORT] ?? null;
+        if (is_null($sort)) {
+            return $next($request->withAttribute(REQUEST_SORT, $handler::getDefaultSort()), $response);
+        }
+
+        $sortParameters = $this->parseSortParameters($sort);
+        $sortFields = array_column($sortParameters, 0);
```

Comment:
Would it be more readable to return a hash rather than a tuple? 

` ['field' => $field, 'direction' => 'ASC']` ?